### PR TITLE
ceph-daemon: Add basic mypy support

### PR DESCRIPTION
**Pull request by toabctl**
Add mypy support and some type hints to be able to do static code
analysis.

Signed-off-by: Thomas Bechtold <tbechtold@suse.com>

### CODE REVIEW

**Comment by toabctl** [comment]
Note: I stopped to add more type annotations to first get feedback if that is wanted and correctly done.

**Comment by liewegas** [comment]
looks okay to me?

File: src/ceph-daemon/ceph-daemon [hunk 1]
```diff
@@ -191,82 +195,82 @@ def call_throws(command, **kwargs):
...
     data_dir_base = os.path.join(args.data_dir, fsid)
     makedirs(data_dir_base, uid, gid, DATA_DIR_MODE)
     makedirs(os.path.join(data_dir_base, 'crash'), uid, gid, DATA_DIR_MODE)
     makedirs(os.path.join(data_dir_base, 'crash', 'posted'), uid, gid,
              DATA_DIR_MODE)
     return data_dir_base
 
-def make_data_dir(fsid, daemon_type, daemon_id, uid=None, gid=None):
-    if not uid:
+def make_data_dir(fsid, daemon_type, daemon_id, uid=None, gid=None): # type: (str, str, int, int, int) -> str
```
Comment (mgfritch) [inline src/ceph-daemon/ceph-daemon]: Maybe avoid very long lines by putting the type hints on the line below?

```suggestion
def make_data_dir(fsid, daemon_type, daemon_id, uid=None, gid=None):
    # type: (str, str, int, int, int) -> str
```

File: src/ceph-daemon/mypy.ini [hunk 2]
```diff
@@ -0,0 +1,2 @@
+[mypy]
+ignore_missing_imports = True
```
Comment (sebastian-philipp) [inline src/ceph-daemon/mypy.ini]: we have a mypy.ini in `..`` already. Make sense to have only one in ceph?

**Pull request merged**